### PR TITLE
Update the Internationalization tutorial per the optional flutter_localization package

### PR DESCRIPTION
**Pull request by HansMuller**
Sync up the tutorial with the changes made in https://github.com/flutter/flutter/pull/12410

### CODE REVIEW

**Comment by Sfshaza** [comment]
It's odd that Hans' email wasn't recognized by the googlebot.

File: tutorials/international/index.md [hunk 1]
```diff
@@ -25,6 +25,56 @@ supports. Flutter provides widgets and classes that help with
 internationalization and the Flutter libraries themselves are
 internationalized.
 
+### Setting up an Internationalized app: the `flutter_localizations` package
+
+By default Flutter only provides US English localizations. To add
+support for other languages, applications must specify some additional
+`MaterialApp` properties and include a separate package called
```
Comment (sethladd) [inline tutorials/international/index.md]: what if you don't use MaterialApp ?

File: tutorials/international/index.md [hunk 2]
```diff
@@ -25,6 +25,56 @@ supports. Flutter provides widgets and classes that help with
...
+`flutter_localizations`.  This package currently supports about 15
+languages with more to come.
```
Comment (sethladd) [inline tutorials/international/index.md]: don't promise more will come. Just say "we have XX languages as of YY date"

File: tutorials/international/index.md [hunk 3]
```diff
@@ -25,6 +25,56 @@ supports. Flutter provides widgets and classes that help with
...
+
+International apps must add a `pubspec.yaml` dependency on flutter_localizations:
```
Comment (sethladd) [inline tutorials/international/index.md]: maybe better as "To use flutter_localizations, add the package a dependency to your pubspec.yaml file"

File: tutorials/international/index.md [hunk 4]
```diff
@@ -25,6 +25,56 @@ supports. Flutter provides widgets and classes that help with
...
+
+```dart
+dependencies:
+  flutter:
+    sdk: flutter
+  flutter_localizations:
+    sdk: flutter
+```
+
+International apps must also add an import for flutter_localizations
```
Comment (sethladd) [inline tutorials/international/index.md]: "Next, import the flutter_localizations library and specify..."

File: tutorials/international/index.md [hunk 5]
```diff
@@ -25,6 +25,56 @@ supports. Flutter provides widgets and classes that help with
...
+ supportedLocales: [
+    const Locale('en', 'US'), // English
+    const Locale('he', 'IL'), // Hebrew
+    // ... other locales the app supports
+  ],
+  // ...
+)
+```
+
+The elements of `localizationsDelegates` are factories that produce
```
Comment (sethladd) [inline tutorials/international/index.md]: what do you mean "elements" here?

classes?

File: tutorials/international/index.md [hunk 6]
```diff
@@ -25,6 +25,56 @@ supports. Flutter provides widgets and classes that help with
...
+collections of localized values. `GlobalMaterialLocalizations.delegate`
+provides localized strings and other values for the materials
```
Comment (sethladd) [inline tutorials/international/index.md]: materials is plural ?

File: tutorials/international/index.md [hunk 7]
```diff
@@ -100,6 +150,99 @@ static MaterialLocalizations of(BuildContext context) {
...
+package includes implementations of the `MaterialLocalizations` and
+`WidgetsLocalizations` interfaces that only provide US English
+values. These implementation classes are called
+`DefaultMaterialLocalizations` and
+`DefaultWidgetsLocalizations`. They're included automatically unless a
+different delegate of the same base type is specified with the app's
+`localizationsDelegates` parameter.
+
+The flutter_localizations package includes multi-language
+    implementations of the localizations interfaces called `GlobalMaterialLocalizations`
```
Comment (sethladd) [inline tutorials/international/index.md]: nit: spaces at beginning of line

File: tutorials/international/index.md [hunk 8]
```diff
@@ -100,6 +150,99 @@ static MaterialLocalizations of(BuildContext context) {
...
+)
+```
+
+The global localization delegates construct locale-specific instances
+of the corresponding classes. For example
+`GlobalMaterialLocalizations.delegate` is a `LocalizationsDelegate`
+that produces an instance of `GlobalMaterialLocalizations`.
+
+Currently the global localization classes support about 15 langauges
+(more are coming). Applications can add support for additional
```
Comment (sethladd) [inline tutorials/international/index.md]: be specific about num of languages, add "as of YY date", and remove promise that more are coming. Even better, link to the directory that contains all the languages.

File: tutorials/international/index.md (hunk 1 above)
Comment (HansMuller) [reply tutorials/international/index.md]: I added a paragraph about that below the example.

File: tutorials/international/index.md (hunk 5 above)
Comment (HansMuller) [reply tutorials/international/index.md]: localizationsDelegates is a List, so list elements.

File: tutorials/international/index.md (hunk 6 above)
Comment (HansMuller) [reply tutorials/international/index.md]: No. Widgets library, material library. I've fixed that.

File: tutorials/international/index.md (hunk 1 above)
Comment (HansMuller) [reply tutorials/international/index.md]: I've also added qualifying paragraph in the opening section.

**Comment by sethladd** [comment]
Lgtm

**Pull request merged**